packages/field-base/src/input-control-mixin.js

```diff
@@ -23,6 +23,22 @@ export const InputControlMixin = (superclass) =>
   ) {
     static get properties() {
       return {
+        /**
+         * A pattern matched against individual characters the user inputs.
+         *
+         * When set, the field will prevent:
+         * - `keydown` events if the entered key doesn't match `/^allowedCharPattern$/`
+         * - `paste` events if the pasted text doesn't match `/^allowedCharPattern*$/`
+         * - `drop` events if the dropped text doesn't match `/^allowedCharPattern*$/`
+         *
+         * For example, to allow entering only numbers and minus signs, use:
+         * `allowedCharPattern = "[\\d-]"`
+         */
+        allowedCharPattern: {
```

Comment:
For some reason the following doesn't render at all (added on text-field dev page):
```html
<vaadin-text-field label="Allowed char pattern" required allowed-char-pattern="[a]*"></vaadin-text-field>
```

Reply:
Good catch, added the `try` / `catch` and logic to use `console.error` instead of throwing an exception.

Reply:
Agree, we could catch this error and maybe log it.